<?xml version="1.0"?>
<launch>
    <arg name="only_line_follow"    default="false"/>
    <include file="$(find jetauto_peripherals)/launch/astrapro.launch"/>
    <include file="$(find hiwonder_servo_controllers)/launch/start.launch"/>
    <include file="$(find jetauto_controller)/launch/jetauto_controller.launch"/>

    <arg name="node_name" default="yolov5_node" />
    <rosparam param="/$(arg node_name)/classes">['go', 'right', 'park', 'red', 'green', 'crosswalk']</rosparam>
    <node unless="$(arg only_line_follow)" pkg="jetauto_example" type="yolov5_node.py" name="$(arg node_name)" output="screen">
        <param name="use_depth_cam" value="true"/>
        <param name="engine"        value="traffic_signs_640s_7_0.engine"/>
        <param name="lib"           value="libmyplugins_640.so"/>
        <param name="conf_thresh"   value="0.8"/>
    </node>

    <node pkg="jetauto_example" type="self_driving.py" name="self_driving" output="screen">
          <param name="only_line_follow"    value="$(arg only_line_follow)"/>
    </node>
</launch>
